files/check_keepalive.rb

```diff
@@ -0,0 +1,97 @@
+module Sensu
+module Extension
+
+# 1) update per-check timestamps in redis
+# 2) have eventmachine periodic timer to check if anything timed out and
+#    generate a new "critical" event for every
+#
+# extension has access to:
+#  - logger     @logger
+#  - redis      @settings[:redis]
+#  - transport  @settings[:transport]
+
+class CheckKeepalive < Handler
+  def name
+    'check_keepalive'
+  end
+
+  def description
+    'emit per-check keepalive events'
+  end
+
+  def post_init
+    @redis    = @settings.redis
+    @em_timer = EventMachine::PeriodicTimer.new(10) { process_keepalives }
+  end
+
+  def stop
+    @em_timer.cancel
+    yield
+  end
+
+  def run(event)
+    client_name = event[:client][:name]
+    check_name  = event[:check][:name]
+
+    rkey = "ext:check_keepalive:checks"
+    hkey = "#{client_name}:#{check_name}"
+    hval = event[:check][:keepalive]
```

Comment:
followup to https://github.com/Yelp/puppet-monitoring_check/pull/33#discussion_r24506358

@solarkennedy No, this means that puppet has removed the check. No alert necessary.

sensu server is supposed to alert when it doesn't hear about a check we asked to monitor for keepalive. how does sensu-server know if check went silent because it was removed by puppet or because there is a problem with running the check?

imo ensure => absent on a keepalived puppet-scheduled check should notify sensu-server about it by either sending a special api request or dropping the check keepalive info from redis or (and this is sort of nice) sending an OK with keepalive=nil so that check is auto-removed from this extension's keepalive-hash. am i making sense at all?